src/Stryker.Core/Stryker.Core/DiffProviders/GitDiffProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿using LibGit2Sharp;
+using Microsoft.Extensions.Logging;
+using Stryker.Core.Exceptions;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using System.Collections.ObjectModel;
+using System.IO;
+
+namespace Stryker.Core.DiffProviders
+{
+    public class GitDiffProvider : IDiffProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly ILogger _logger;
+
+        public GitDiffProvider(StrykerOptions options)
+        {
+            _options = options;
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<GitDiffProvider>();
+        }
+
+        public DiffResult ScanDiff()
+        {
+            if (!_options.DiffEnabled)
```

Comment:
If we don't do this check, the options don't need to be passed to this class. The caller should decide whether or not to call the scandiff